src/core_plugins/kibana/public/dashboard/dashboard_state.js

```diff
@@ -128,33 +325,41 @@ export class DashboardState {
    * @param toJson {function} A custom toJson function. Used because the previous code used
    * the angularized toJson version, and it was unclear whether there was a reason not to use
    * JSON.stringify
+   * @param timefilter
    * @returns {Promise<string>} A promise that if resolved, will contain the id of the newly saved
    * dashboard.
    */
-  saveDashboard(toJson) {
+  saveDashboard(toJson, timeFilter) {
```

Comment:
With the whole move to React, I haven't heard much discussion about adopting a 'Flux' architecture, but this would potentially change the way that all of this is structured. The 'Flux' architecture doesn't really fit with the ActiveRecord inspired pattern of the SavedObjects, and I'm not sure what overlap the Stores would have with the AppState itself.

I'm fine with leaving it here for now, no reason to refactor a bunch of stuff that will change in the future, and what you've done makes this dramatically more readable.